src/tye/Program.DefaultOptionsMiddleware.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.Linq;
+
+namespace Microsoft.Tye
+{
+    public static partial class Program
+    {
+        private const string DefaultOptionsEnvVarPrefix = "TYE_";
+        private const string DefaultOptionsEnvVarPostfix = "_ARGS";
+
+        private static void DefaultOptionsMiddleware(InvocationContext context)
+        {
+            if (context.ParseResult.CommandResult.Parent != context.ParseResult.RootCommandResult)
```

Comment:
Is this a common check for these types of middleware? May be worth a comment why you check that.

Reply:
Well, not exactly.
I'm checking if the currently executing command is second level (probably not official name but i'm not sure how to name it) command. That means that the command is subcommand of root.
For example:
 - Input `tye run` is described like this
    - `tye` is root command
    - `run` is second level command
 - Input `tye build`
    - `tye` is root command
    - `build` is second level command

This check is not necessary at the moment. I was just trying to be safe for the future.
If there will be added third level command and the name of the command will be same as one of the second level (for example `tye build run` - `run` is third level and also second level command), then it may add unintended options. So i rather disable it for the moment.

If you know that there will not be third level commands, then i can remove this chcek. 
Othervise i will put some comment there.